File: PRI2/solutions/sol.py
N, S = map(int, input().split())

# Process all swaps
pea_loc = S
for i in range(N):
    a, b = map(int, input().split())
    if pea_loc == a:
        pea_loc = b
    elif pea_loc == b:
        pea_loc = a

print(pea_loc)

File: PRI2/solutions/sol.cpp
#include <iostream>

using namespace std;

int main() {
    int N, S;
    cin >> N >> S;
    
    // Process all swaps
    int peaLoc = S;
    for (int i = 0; i < N; i++) {
        int a, b;
        cin >> a >> b;
        if (peaLoc == a) {
            peaLoc = b;
        } else if (peaLoc == b) {
            peaLoc = a;
        }
    }

    cout << peaLoc << endl;
    return 0;
}